### PR TITLE
Add variants of octothorpe (the '#' character)

### PR DESCRIPTION
**Pull request by bl-ue**
https://en.wiktionary.org/wiki/octothorpe

### CODE REVIEW

File: codespell_lib/data/dictionary.txt [hunk 1]
```diff
@@ -19550,6 +19550,28 @@ octect->octet
 octohedra->octahedra
 octohedral->octahedral
 octohedron->octahedron
+octothhorpes->octothorpes
+octothorp->octothorpe
+octothorpe->octothorpe
```
Comment (peternewman) [inline codespell_lib/data/dictionary.txt:19555]: This one just corrects to itself!

**Review by peternewman — CHANGES_REQUESTED** [review]
Are these actual or generated?

File: codespell_lib/data/dictionary.txt [hunk 2]
```diff
@@ -19550,6 +19550,28 @@ octect->octet
...
+octothrpes->octothorpes
+octotohrpe->octothorpe
+octotohrpes->octothorpes
+octotorpe->octothorpe
+octotorpes->octothorpes
+octotrhope->octothorpe
+octotrhopes->octothorpes
+octotthhorpe->octothorpe
+octotthhorpes->octothorpes
+octotthorpes->octothorpe
```
Comment (peternewman) [inline codespell_lib/data/dictionary.txt:19573]: This should probably be removed!

**Comment by bl-ue** [comment]
>Are these actual or generated?

Both, in a way. 🙂  I manually typed some likely typos as this is rather long word, and then I used VS Code multi-cursor to mess them up a bit more.

>This should probably be removed!

Yeah, they're rather silly I admit.  There were actually more before I opened PR but I trimmed some down... 😄

File: codespell_lib/data/dictionary.txt (hunk 1 above)
Comment (bl-ue) [inline codespell_lib/data/dictionary.txt:19555]:
```suggestion
```

File: codespell_lib/data/dictionary.txt (hunk 2 above)
Comment (bl-ue) [inline codespell_lib/data/dictionary.txt:19573]:
```suggestion
```

Comment (bl-ue) [inline codespell_lib/data/dictionary.txt:19573]:
```suggestion
```

File: codespell_lib/data/dictionary.txt [hunk 3]
```diff
@@ -19550,6 +19550,24 @@ octect->octet
 octohedra->octahedra
 octohedral->octahedral
 octohedron->octahedron
+octothhorpes->octothorpes
```
Comment (peternewman) [inline codespell_lib/data/dictionary.txt:19553]: We should at least have the singular

**Comment by peternewman** [comment]
> > Are these actual or generated?
> 
> Both, in a way. slightly_smiling_face I manually typed some likely typos as this is rather long word, and then I used VS Code multi-cursor to mess them up a bit more.

I'm not sure we should necessarily be flooding the dictionary with random potential typos, it would be nice to see some real world examples that some of these exist as a base, even if we then add the plural etc for example.

**Comment by bl-ue** [comment]
Okay, I'll keep the simple obvious ones and get rid of the less likely ones.  How about that?

**Comment by peternewman** [comment]
> Okay, I'll keep the simple obvious ones and get rid of the less likely ones. How about that?

In case my :+1: wasn't explicit enough, that sounds good to me.

**Comment by peternewman** [comment]
What happened @bl-ue ? I thought you were going to just sort the simple/obvious ones?

**Comment by bl-ue** [comment]
Since I opened this PR, I realized that it's better to provide known typos, rather than possible ones.  None of these I've ever actually seen, so I figured it's just a waste of time.  There's much more interesting work that can be done in terms of adding to the database :wink:

**Pull request closed without merging**